lib/semian/lru_hash.rb

```diff
@@ -0,0 +1,53 @@
+class LRUHash
+  extend Forwardable
+  def_delegators :@table, :size, :count, :empty?, :keys, :clear, :delete
+
+  attr_reader :table, :minimum_time_in_lru
+  def initialize(minimum_time_in_lru = 300)
+    @minimum_time_in_lru = minimum_time_in_lru
```

Comment:
The problem I was getting and that got fixed by adding that, is this test: https://github.com/Shopify/semian/blob/master/test/protected_resource_test.rb#L96

We register the resource, but at this point, there is no error in the window, which will cause the clean_unused_resource method to remove it as soon as it gets there. I feel like I might be missing something or maybe this should not happen.

Reply:
Yeah, thinking more about it I think it's reasonable to have this. I think the heuristic here really is that it only really makes sense to have a circuit if you have an error.